packages/compiler/test/shadow_css/keyframes_spec.ts

```diff
@@ -0,0 +1,498 @@
+  });
+
+  it('should handle the usage of quotes', () => {
+    const css = `
+        div {
+            animation: 1.5s foo;
+        }
+
+        p {
+            animation: 1s 'foz bar';
+        }
+
+        @keyframes 'foo' {}
+        @keyframes "foz bar" {}
+        @keyframes bar {}
+        @keyframes baz {}
+        @keyframes qux {}
+        @keyframes quux {}
+        `;
+    const result = s(css, 'host-a');
+    expect(result).toContain('@keyframes \'host-a_foo\' {}');
+    expect(result).toContain('@keyframes "host-a_foz bar" {}');
+    expect(result).toContain('animation: 1.5s host-a_foo;');
+    expect(result).toContain('animation: 1s \'host-a_foz bar\';');
+  });
+
+  it('should handle the usage of quotes containing escaped quotes', () => {
+    const css = `
+        div {
+            animation: 1.5s "foo\\"bar";
+        }
+
+        p {
+            animation: 1s 'bar\\' \\'baz';
+        }
+
+        button {
+            animation-name: 'foz " baz';
+        }
+
+        @keyframes "foo\\"bar" {}
+        @keyframes "bar' 'baz" {}
+        @keyframes "foz \\" baz" {}
+        `;
+    const result = s(css, 'host-a');
+    expect(result).toContain('@keyframes "host-a_foo\\"bar" {}');
+    expect(result).toContain(`@keyframes "host-a_bar' 'baz" {}`);
+    expect(result).toContain(`@keyframes "host-a_foz \\" baz" {}`);
+    expect(result).toContain('animation: 1.5s "host-a_foo\\"bar";');
+    expect(result).toContain(`animation: 1s 'host-a_bar\\' \\'baz';`);
+    expect(result).toContain(`animation-name: 'host-a_foz " baz';`);
+  });
+
+  it('should handle the usage of commas in multiple animation definitions in a single declaration',
+     () => {
+       const css = `
+        button {
+        animation: 1s "foo bar, baz", 2s 'qux quux';
+        }
+
+        div {
+        animation: 500ms foo, 1s 'bar, baz', 1500ms bar;
+        }
+
+        p {
+        animation: 3s "bar, baz", 3s 'foo, bar' 1s, 3s "qux quux";
+        }
+
+        @keyframes "qux quux" {}
+        @keyframes "bar, baz" {}
+    `;
+       const result = s(css, 'host-a');
+       expect(result).toContain('@keyframes "host-a_qux quux" {}');
+       expect(result).toContain('@keyframes "host-a_bar, baz" {}');
+       expect(result).toContain('animation: 1s "foo bar, baz", 2s \'host-a_qux quux\';');
+       expect(result).toContain(`animation: 500ms foo, 1s 'host-a_bar, baz', 1500ms bar;`);
+       expect(result).toContain(
+           `animation: 3s "host-a_bar, baz", 3s 'foo, bar' 1s, 3s "host-a_qux quux";`);
+     });
+
+  it('should handle the usage of double quotes escaping in multiple animation definitions in a single declaration',
+     () => {
+       const css = `
+        div {
+            animation: 1s "foo", 1.5s "bar";
+            animation: 2s "fo\\"o", 2.5s "bar";
+            animation: 3s "foo\\"", 3.5s "bar", 3.7s "ba\\"r";
+            animation: 4s "fo\\\"o", 4.5s "bar", 4.7s "baz\\"";
```

Comment:
I don't think the triple slash does what you intend it to do. Since this is already inside a string, `\\\"` is equivalent to `\"` (outside a string). So, the third slash is basically escaping `"` inside the backtick string (which is a no-op).

You probably meant to have something like:

```suggestion
            animation: 4s "foo\\\\", 4.5s "bar", 4.7s "baz\\"";
            animation: 5s "fo\\\\\\"o", 5.5s "bar";
```

(Same for the following two tests.)

Reply:
yeah, you're totally right, sorry my mistake! :sweat:

Reply:
damn, sorry I missed the last line of your comment, and indeed thought I fixed this one! sorry!!!!! :bow:

Reply:
Hopefully should be all ok now :slightly_smiling_face: